.github/workflows/benchmark.yml

```
@@ -0,0 +1,64 @@
+---   # Proof generation benchmarking workflow
+
+name: Benchmark proving
+
+on:
+  schedule:
+    - cron: "0 4 * * *"
+  workflow_dispatch:
+    branches:
+      - "**"
+
+env:
+  BENCHMARK_WITNESS: artifacts/witness_b1000_b1019.json
+
+jobs:
+  benchmark_proving:
+    name: Benchmark proving for representative blocks
+    runs-on: zero-reg
+    timeout-minutes: 900  # Timeout TBD
```

```suggestion
    timeout-minutes: 900  # Timeout TBD
```
nit: TBD in this PR, or later? Add a tracking issue if needed (or simply remove the comment ;))

---

Set by @Nashtare just as a precaution. Benchmark should be between 2.5 hours according to last run.
I could lower the number a bit

---

I think we can just remove the timeout, whittling it down is a future task